### PR TITLE
Autobuild Function calling

### PR DESCRIPTION
**Pull request by krishnashed**
## Why are these changes needed?

Introduction of Autobuild, didn't support function/tool calling. This PR is solution to add this feature/

## Related issue number

Closes https://github.com/microsoft/autogen/issues/1510

### CODE REVIEW

File: test/agentchat/contrib/test_agent_builder.py [hunk 1]
```diff
@@ -69,6 +90,99 @@ def test_build():
     assert len(agent_config["agent_configs"]) <= builder.max_agents
 
 
+@pytest.mark.skipif(skip_openai, reason=reason + "OR dependency not installed")
```
Comment (sonichi) [inline test/agentchat/contrib/test_agent_builder.py:93]:
```suggestion
@pytest.mark.skipif(skip_openai or skip, reason=reason + "OR dependency not installed")
```

**Review by sonichi — COMMENTED** [review]
Two tests are still failing.

**Comment by rysweet** [comment]
This PR is against AutoGen 0.2. AutoGen 0.2 has been moved to the 0.2 branch. Please rebase your PR on the 0.2 branch or update it to work with the new AutoGen 0.4 that is now in main.

**Comment by krishnashed** [comment]
Ya sure, let me get back to you

**Review by rysweet — APPROVED** [review]

**Pull request merged**